Lib/test/test_dataclasses.py

```diff
@@ -2767,6 +2767,21 @@ class C:
             c.i = 5
         self.assertEqual(c.i, 10)
 
+    def test_frozen_empty(self):
+        @dataclass(frozen=True)
+        class C:
+            pass
+
+        c = C()
+        self.assertFalse(hasattr(c, 'i'))
+        with self.assertRaises(FrozenInstanceError):
+            c.i = 5
```

Comment:
Can we also test that you _can_ add an attribute to a non-dataclass subclass of the empty frozen dataclass?